packages/mui-material/src/styles/ThemeProvider.test.tsx

```diff
@@ -15,4 +41,56 @@ describe('ThemeProvider', () => {
       ),
     ).not.toWarnDev();
   });
+
+  describe('light & dark', () => {
+    function ModeSwitcher() {
+      const { mode, setMode } = useColorScheme();
+      if (!mode) {
+        return null;
+      }
+      return (
+        <select
+          data-testid="mode-switcher"
+          value={mode}
+          onChange={(event) => {
+            setMode(event.target.value as 'light' | 'dark' | 'system');
+          }}
+        >
+          <option value="system">System</option>
+          <option value="light">Light</option>
+          <option value="dark">Dark</option>
+        </select>
+      );
+    }
+
+    it('should be able to use `useColorScheme`', () => {
+      const theme = createTheme({
+        colorSchemes: { dark: true },
+      });
+      expect(() =>
+        render(
+          <ThemeProvider theme={theme}>
+            <ModeSwitcher />
+          </ThemeProvider>,
+        ),
+      ).not.toErrorDev();
+    });
+
+    it('should be able to switch between modes', () => {
+      const theme = createTheme({
+        colorSchemes: { dark: true },
+      });
+      const { getByTestId } = render(
+        <ThemeProvider theme={theme}>
+          <ModeSwitcher />
+        </ThemeProvider>,
+      );
+
+      expect(getByTestId('mode-switcher')).to.have.property('value', 'system');
+
+      fireEvent.change(getByTestId('mode-switcher'), { target: { value: 'dark' } });
```

Comment:
Could we use [userEvent](https://testing-library.com/docs/user-event/intro/) (returned from [render](https://github.com/mui/material-ui/blob/next/packages-internal/test-utils/src/createRenderer.tsx#L303) as `user`)?

Reply:
Done.